packages/cli/src/index.ts

```diff
@@ -22,6 +22,11 @@ module.exports = (): void => {
   const log = new Log('cli');
   const args = process.argv.slice(2);
   cli.run(args)
+      .then((result) => {
+        if (!result) {
+          process.exit(1);
+        }
+      })
       .catch((err: Error) => {
         log.error(err.message);
         process.exit(1);
```

Comment:
I'm not quite sure what you're doing here. Isn't this the last code to be called in the execution of the program? Why would you need to call process.exit() ? Is it just to make the return code 1?

I'm dubious about using process.exit in general since it's not obvious from the caller of this method that the entire program could terminate. Is there some way to make the method return the exit code - so 1 in the two cases you call `process.exit` in and 0 otherwise?

Reply:
`cli/src/index.ts` is the entry point for the CLI application - a "main" method (TypeScript doesn't have a main). The reason it calls process.exit(1) is to signal an abnormal termination. In case the program is being invoked as part of a build pipeline, this signals that the CLI failed and the whole pipeline can be aborted.

Reply:
Could we instead do:

    const success = await cli.run(args)
        .then((result) => {
            return result;
        })
        .catch((error) => {
            // ...
            return false;
        });
    process.exit(success ? 0 : 1);

Reply:
Updated the code.